### ISSUE TITLE
Documentation Page - Proposal & Ideas

### ISSUE THREAD

**Issue by georgelemon** [vlang/vinix-website#2]
I think the website needs a CSS framework instead of plain. Personally, I always use Bootstrap (only CSS, no JS).

I can help with the following:

- Search functionality backed by IndexedDB and [Autocomplete.js](https://tarekraafat.github.io/autoComplete.js/demo/)
- Lightweight Theme SCSS & Bootstrap based
- JS functionality (minified by Rollup):
- Integrate with WebWorkers API for Offline usage

<img width="1525" alt="Screenshot 2021-08-01 at 8 53 30 AM" src="https://user-images.githubusercontent.com/1230420/127760919-13cf7077-0e9f-461f-94c1-d855e34ed372.png">

<img width="1525" alt="Screenshot 2021-08-01 at 10 25 36 AM" src="https://user-images.githubusercontent.com/1230420/127762928-4fba3b79-3c90-4456-aba5-d3b267a33d8f.png">

**Comment by georgelemon** [comment]
Ok, I just discovered that V has a documentation parser based on docstrings and also there is a [Github action for V projects](https://github.com/marketplace/actions/create-documentation-for-v-modules)

**Comment by medvednikov** [comment]
it's nice, it's a bit too generic, but if it allows us to get a website quickly, why not

**Comment by Gladear** [comment]
I think it's great for a documentation website to be generic. It's easier for users to grasp, and therefore makes them want to stay.

I just disagree about the technical choices:
- I think Bootstrap will only get in the way. I tend to use Bootstrap less and less, since I usually don't use much more than 3 or 4 components, that I have to tweak anyway, so it's almost the same as rewriting same from scratch. Also SCSS is very much overkill for any simple website in my opinion
- Adding an autocomplete feature using a library should take 20 - 30 lines of JavaScript, using a bundler probably isn't necessary
- We could add a service worker... But it definitely isn't a priority

**Comment by georgelemon** [comment]
@Gladear, I agree but take in consideration that since Bootstrap sources are available in SCSS we can grab from there only the utilities, form and grid system. Also, the Autocomplete.js library has 650 lines unminified (no external deps), so minified is around 9kb.

**Comment by JalonSolov** [comment]
A baseline assumption: Whatever is used, it should be as simple and self-contained as possibly, and should NOT require any complex build system (which includes node.js/npms, etc.).  It should also be as small as possible, so that load times are snappy.  We don't want 100k+ of JS+CSS for a simple landing page.
